imagepipeline/src/main/java/com/facebook/imagepipeline/request/ImageRequestBuilder.java

```diff
@@ -47,6 +56,34 @@ public static ImageRequestBuilder newBuilderWithSource(Uri uri) {
     return new ImageRequestBuilder().setSource(uri);
   }
 
+  /**
+   * <p/>
+   * Only image resources can be used with the image pipeline (e.g. a PNG image). Other resource
+   * types such as Strings or XML Drawables make no sense in the context of the image pipeline and
+   * so cannot be supported by definition. One potentially confusing case is drawable declared in
+   * XML (e.g. ShapeDrawable). Important thing to note is that this is not an image. If you want
+   * to display an XML drawable as the main image, then set it as a placeholder and use the null uri.
+   * <p/>
+   * See also : {@link com.facebook.common.util.UriUtil#isLocalResourceUri(android.net.Uri)}
+   *
+   * @param resources application resources.
+   * @return a new request builder instance.
+   */
+    TypedValue value = new TypedValue();
+    resources.getValue(resId, value, true);
+    String resourceName = value.string.toString();
+    String resourceExt = resourceName.substring(resourceName.lastIndexOf('.'));
+
+    if (!resourceExt.equals(LOCAL_RESOURCE_EXT_PNG) && !resourceExt.equals(LOCAL_RESOURCE_EXT_JPG)) {
+      throw new IllegalArgumentException("Only image resources can be used with the image"
```

Comment:
`"Only image resources can be used with the image pipeline. " + resourceExt + " files are not supported."`